packages/server/src/subscriptions/websockets.test.ts

```diff
@@ -691,9 +691,26 @@ describe('WebSocket Subscription', () => {
           await sleep(0);
         });
 
-      expect(globalLoggerErrorSpy).toHaveBeenCalledWith('[WS] Error occurred while rewriting attachments', {
-        err: expect.any(Error),
-      });
+      let shouldFail = false;
+      const failTimer = setTimeout(() => {
+        shouldFail = true;
+      }, 5000);
```

Comment:
It might be simpler to do this by recording `const startTime = Date.now()` above the loop, and then do `while (Date.now() - startTime < 5000) { ... }`